workflow-job

```diff
@@ -608,8 +608,14 @@ def buildVolumeFromTag(def node) {
     sh "docker run ${verbose} -t --name ${containers.gitPublish} -v /origin ${images.gitPublish}"
 }
 
-def compile() {
-    sh "docker build --tag=${images.jamesCompile} dockerfiles/compilation/java-8"
+def compile(def node) {
+    if (node.fileExists('dockerfiles/compilation/java-8')) {
+        javaVersion = java-8
+    } else {
+        javaVersion = java-11
+    }
+
+    sh "docker build --tag=${images.jamesCompile} dockerfiles/compilation/${javaVersion}"
```

Comment:
I'm not sure that the `javaVersion` variable is visible here.

Isn't it visible only inside the `if` & `else` blocks?